app/containers/ItunesContainer/index.js

```diff
@@ -0,0 +1,100 @@
+import React, { useEffect, useState, memo } from 'react';
+import PropTypes from 'prop-types';
+import { get } from 'lodash';
+import debounce from 'lodash/debounce';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import { itunesContainerCreators } from './reducer';
+import saga from './saga';
+import { injectIntl } from 'react-intl';
+import { selectTrackName, selectTracksData, selectTracksError } from './selectors';
+import { injectSaga } from 'redux-injectors';
+import { compose } from 'redux';
+import isEmpty from 'lodash/isEmpty';
+
+export function ItunesContainer({ dispatchItuneTracks, dispatchClearItuneTracks, tracksData, trackName }) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(tracksData, 'results', null);
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [tracksData]);
+
+  useEffect(() => {
+    if (trackName && !tracksData?.results?.length) {
+      dispatchItuneTracks(trackName);
+      setLoading(true);
+    }
+  }, []);
+
+  const handleOnChange = (tName) => {
+    if (!isEmpty(tName)) {
+      dispatchItuneTracks(tName);
```

Comment:
can we handle this in the useEffect.

Reply:
Sure, made the changes.

Reply:
Instead on calling the dispatch func here, we can use a state and change it which would trigger useEffect to make the dispatch call.

Reply:
can you try to do it in useEffect